File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.ts
function minimumLength(s: string): number {
    let l = 0, r = s.length - 1;
    while (l < r && s[l] === s[r]) {
        let ch = s[l];
        while (l <= r && s[l] === ch) {
            l += 1;
        }
        while (l <= r && s[r] === ch) {
            r -= 1;
        }
    }
    return (l > r) ? 0 : r - l + 1;
};

File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.java
class Solution {
    public int minimumLength(String s) {
        int l = 0, r = s.length() - 1;
        while (l < r && s.charAt(l) == s.charAt(r)) {
            char ch = s.charAt(l);
            while (l <= r && s.charAt(l) == ch) {
                l += 1;
            }
            while (l <= r && s.charAt(r) == ch) {
                r -= 1;
            }
        }
        return (l > r) ? 0 : r - l + 1;
    }
}